llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -78,6 +78,12 @@ class SIFoldOperandsImpl {
   bool frameIndexMayFold(const MachineInstr &UseMI, int OpNo,
                          const MachineOperand &OpToFold) const;
 
+  /// Fold %vgpr = COPY (S_ADD_I32 x, frameindex)
+  ///
+  ///   => %vgpr = V_ADD_U32 x, frameindex
```

Comment:
What prevents it from constant bus violation?

Reply:
Frame index is treated as if it were a VGPR since a few months ago, so we don't need to care what the other operand is